release.sh

```diff
@@ -22,26 +43,38 @@ fi
 
 OLD_VERSION="$(xmlstarlet sel -t -v /_:project/_:version pom.xml | sed -e s/-SNAPSHOT//g)"
 
-# Get new version from user
-echo "${OLD_VERSION}" >version.temp
-${EDITOR:-editor} version.temp
-NEW_VERSION="$(cat version.temp)"
-rm version.temp
+# Get new version from changes
+MAJOR=$(echo $OLD_VERSION | cut -d . -f 1 -)
+MINOR=$(echo $OLD_VERSION | cut -d . -f 2 -)
+PATCH=$(echo $OLD_VERSION | cut -d . -f 3 -)
+
+if [[ "${RELEASE_TYPE}" = "major" ]]; then
+  MAJOR=$((MAJOR+1))
+  MINOR=0
+  PATCH=0
+elif [[ "${RELEASE_TYPE}" = "minor" ]] || [[ -n $(find "${CHANGE_DIR}" -mindepth 1 -maxdepth 1 \
+    -name "add_*" -or -name "change_*" -or -name "remove_*") ]]; then
+  RELEASE_TYPE="minor"
+  MINOR=$((MINOR+1))
+  PATCH=0
+elif [[ -n $(find "${CHANGE_DIR}" -mindepth 1 -maxdepth 1 -name "fix_*") ]]; then
+  RELEASE_TYPE="patch"
+  # use patch version from snapshot version
+else
+  echo "No changes found in 'changes' directory. Aborting release." >&2
+  exit 5
+fi
+
+NEW_VERSION="${MAJOR}.${MINOR}.${PATCH}"
 
 if git tag --list | grep -c -E "^v${NEW_VERSION}$" >/dev/null; then
 	echo "Version $NEW_VERSION already exists. Please restart and select a different version number."
 	exit 3
 fi
 
 # Update the release notes
-if ! grep -q "# Unreleased" RELEASE_NOTES.md; then
-	echo "Error: changelog does not contain an Unreleased section"
-	exit 4
-fi
-mv RELEASE_NOTES.md RELEASE_NOTES.md.old
-sed "s/# Unreleased/# Unreleased\n\n# \[${NEW_VERSION}\] - $(date -u +%Y-%m-%dT%H:%M+00:00)/g" RELEASE_NOTES.md.old > RELEASE_NOTES.md
-rm RELEASE_NOTES.md.old
-${EDITOR:-editor} RELEASE_NOTES.md
+echo "Preparing ${RELEASE_TYPE} release ${NEW_VERSION}..."
+./compact-changelog.sh ${NEW_VERSION} || return 2
```

Comment:
```suggestion
./compact-changelog.sh ${NEW_VERSION} || exit 2
```
Sorry, missed this on first pass

Reply:
Fixed!